File: easy/0412-FizzBuzz/FizzBuzz.py
from typing import List

class Solution:
    # Time complexity: O(n)
    # Space complexity: O(n)
    def fizzBuzz(self, n: int) -> List[str]:
        list = []
        for i in range(1, n + 1):
            if i % 15 == 0:
                list.append("FizzBuzz")
            elif i % 5 == 0:
                list.append("Buzz")
            elif i % 3 == 0:
                list.append("Fizz")
            else:
                list.append(str(i))
        return list

# test
solution = Solution()
print(solution.fizzBuzz(3))
print(solution.fizzBuzz(5))
print(solution.fizzBuzz(15))

File: easy/0412-FizzBuzz/FizzBuzz.java
import java.util.ArrayList;
import java.util.List;

public class FizzBuzz {
    // Time complexity: O(n)
    // Space complexity: O(n)
    public List<String> fizzBuzz(int n) {
        List<String> answer = new ArrayList<String>();
        for (int i = 1; i <= n; i++) {
            if (i % 15 == 0) {
                answer.add("FizzBuzz");
            } else if (i % 5 == 0) {
                answer.add("Buzz");
            } else if (i % 3 == 0) {
                answer.add("Fizz");
            } else {
                answer.add(Integer.toString(i));
            }
        }
        return answer;
    }

    // test
    public static void main(String args[]) {
        FizzBuzz fizzBuzz = new FizzBuzz();
        System.out.println(fizzBuzz.fizzBuzz(3));
        System.out.println(fizzBuzz.fizzBuzz(5));
        System.out.println(fizzBuzz.fizzBuzz(15));
    }
}
